.github/workflows/build-docker-image-and-binaries.yaml

```diff
@@ -63,6 +63,34 @@ jobs:
               TOFND_PATH="/Users/runner/work/tofnd/tofnd/target/release"
           fi
 
+      - name: build darwin binaries
+        env:
+          SEMVER: ${{ github.event.inputs.tag }}
+        if: matrix.os == 'macos-latest'
+        run: |
+          OS="darwin"
+          ARCH="${{ matrix.arch }}"
+          if [ "$ARCH" == "arm64" ]
+          export LIBRARY_PATH=/opt/homebrew/lib:$LIBRARY_PATH
+          export HOMEBREW_NO_INSTALL_CLEANUP=TRUE
+          export LD_LIBRARY_PATH=/opt/homebrew/lib:$LD_LIBRARY_PATH
+          brew uninstall --ignore-dependencies gmp
+          then
+            ARM_DEPENDENCY=$(brew fetch --force --bottle-tag=arm64_sonoma gmp | grep Downloaded | awk '{print $3}')
+            rustup target add aarch64-apple-darwin
+            cargo build --release --target aarch64-apple-darwin
+            mkdir -p tofndbin
+            mv /Users/runner/work/tofnd/tofnd/target/aarch64-apple-darwin/release/tofnd "./tofndbin/tofnd-$OS-$ARCH-$SEMVER"
+          else
+            export RUSTFLAGS="-L/opt/homebrew/lib"
+            export C_INCLUDE_PATH=/opt/homebrew/include:$C_INCLUDE_PATH
+            export CPLUS_INCLUDE_PATH=/opt/homebrew/include:$CPLUS_INCLUDE_PATH
+            ARM_DEPENDENCY=$(brew fetch --force --bottle-tag=arm64_sonoma gmp | grep Downloaded | awk '{print $3}')
+            brew install "$ARM_DEPENDENCY"
+            cargo install --locked --path .
+            mkdir -p tofndbin
+            mv "/Users/runner/work/tofnd/tofnd/target/release/tofnd" "./tofndbin/tofnd-$OS-$ARCH-$SEMVER"
+          fi
```

Comment:
```suggestion

          export LIBRARY_PATH=/opt/homebrew/lib:$LIBRARY_PATH
          export HOMEBREW_NO_INSTALL_CLEANUP=TRUE
          export LD_LIBRARY_PATH=/opt/homebrew/lib:$LD_LIBRARY_PATH

          brew uninstall --ignore-dependencies gmp
          mkdir -p tofndbin

          if [ "$ARCH" == "arm64" ]; then
            ARM_DEPENDENCY=$(brew fetch --force --bottle-tag=arm64_sonoma gmp | grep Downloaded | awk '{print $3}')

            rustup target add aarch64-apple-darwin
            cargo build --release --target aarch64-apple-darwin

            mv /Users/runner/work/tofnd/tofnd/target/aarch64-apple-darwin/release/tofnd "./tofndbin/tofnd-$OS-$ARCH-$SEMVER"
          else
            export RUSTFLAGS="-L/opt/homebrew/lib"
            export C_INCLUDE_PATH=/opt/homebrew/include:$C_INCLUDE_PATH
            export CPLUS_INCLUDE_PATH=/opt/homebrew/include:$CPLUS_INCLUDE_PATH
            AMD64_DEPENDENCY=$(brew fetch --force --bottle-tag=amd64_sonoma gmp | grep Downloaded | awk '{print $3}')

            brew install "$AMD64_DEPENDENCY"
            cargo install --locked --path .

            mv "/Users/runner/work/tofnd/tofnd/target/release/tofnd" "./tofndbin/tofnd-$OS-$ARCH-$SEMVER"
          fi
```
please add more newlines between different groups of cmds (setting variables, running cmds etc). bash is already not very readable and this makes it even harder.

Also, shouldn't the `else` clause install amd64 dependency and not arm64?

Reply:
> Also, shouldn't the else clause install amd64 dependency and not arm64?

I didn't find an `amd64` dep for the gmp lib (looked [here](https://formulae.brew.sh/api/formula/gmp.json)). But it seems that it's able to compile without specifying an architecture manually. Tested it [here](https://github.com/axelarnetwork/tofnd/actions/runs/9374868322), in https://github.com/axelarnetwork/tofnd/pull/329